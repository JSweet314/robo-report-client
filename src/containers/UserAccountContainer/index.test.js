import React from 'react';
import { shallow } from 'enzyme';
import { UserAccountContainer, mapStateToProps } from './';

describe('UserAccountContainer', () => {
  let wrapper;
  const mockHandleOAuthSignOut = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <UserAccountContainer
        user={{ firstName: 'Jerry' }}
        handleOAuthSignOut={mockHandleOAuthSignOut}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should map user from state to props', () => {
      const mockState = { user: { firstName: 'Jill' } };
      const mapped = mapStateToProps(mockState);

      expect(mapped).toHaveProperty('user', { firstName: 'Jill' });
    });
  });
});
