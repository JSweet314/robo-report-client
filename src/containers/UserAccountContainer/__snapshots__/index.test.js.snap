// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccountContainer should match the snapshot 1`] = `
<div
  className="user-account-container"
>
  <p
    className="user-account-container__p"
  >
    <strong>
      First Name: 
    </strong>
    Jerry
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      Last Name: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      Email: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      phone: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      Phone Type: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      Phone Location: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      Address: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      State: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      City: 
    </strong>
  </p>
  <p
    className="user-account-container__p"
  >
    <strong>
      Zip Code: 
    </strong>
  </p>
  <button
    className="user-account-container__button"
  >
    Edit My Information
  </button>
  <button
    className="user-account-container__button"
    onClick={[Function]}
  >
    Sign Out
  </button>
</div>
`;
